examples/webhook-signing/typescript-node-express/express-ts.ts

```diff
@@ -0,0 +1,76 @@
+import Stripe from 'stripe';
+import express from 'express';
+import bodyParser from 'body-parser';
+import env from 'dotenv';
+
+env.config();
+
+const stripe: Stripe = new Stripe(process.env.STRIPE_SECRET_KEY, {
+  apiVersion: '2019-12-03',
+  typescript: true,
+});
+
+const webhookSecret: string = process.env.STRIPE_WEBHOOK_SECRET;
+
+const app: express.Application = express();
+
+// Only use the raw body parser for webhooks.
+app.use(
+  (
+    req: express.Request,
+    res: express.Response,
+    next: express.NextFunction
+  ): void => {
+    if (req.originalUrl === '/webhook') {
+      next();
+    } else {
+      bodyParser.json()(req, res, next);
+    }
+  }
+);
+
+// Stripe requires the raw body to construct the event
+app.post(
+  '/webhook',
+  bodyParser.raw({type: 'application/json'}),
+  (req: express.Request, res: express.Response): void => {
+    const sig: string | string[] = req.headers['stripe-signature'];
+
+    let event: Stripe.Event;
+
+    try {
+      event = stripe.webhooks.constructEvent(req.body, sig, webhookSecret);
+    } catch (err) {
+      // On error, return the error message
+      res.status(400).send(`Webhook Error: ${err.message}`);
+      return;
+    }
+
+    // Do something with event
+    console.log('Success:', event.id);
+
+    // Cast event data to Stripe object
+    let stripeObject: Stripe.PaymentIntent | Stripe.Charge;
```

Comment:
this we should not do... I think better to just do this:

```ts
    switch (event.type) {
      case 'payment_intent.succeeded':
        const pi = event.data.object as Stripe.PaymentIntent;
        console.log(`PaymentIntent status: ${pi.status}`);
        break;
      case 'charge.succeeded':
        const charge = event.data.object as Stripe.Charge;
        console.log(`Charge id: ${charge.id}`);
        break;
    }
```

Reasoning: 
- sharing the variable could make it easier for different switch cases to affect each other or to do weird things after the switch
- having a union (`|`) implies that _anywhere_ you use `stripeObject`, it could be either a PI or a Charge, whereas  reality is a little different. Also, users could forget to update the union even when handling more event types, which could be misleading

Reply:
Fair, problem is if I move the const declaration into the switch case the linter complains with `Unexpected lexical declaration in case block. eslint(no-case-declarations)`. I'll refactor to if else.

Reply:
Nice, yeah, +1 to if/else.